### PR TITLE
feat: include TemplateRef in MUR assertions

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: pkg/test/masteruserrecord/assertion.go [hunk 1]
```diff
@@ -171,6 +172,7 @@ TierNamespaces:
 		for _, uaNs := range userAccount.Spec.NSTemplateSet.Namespaces {
 			if ns.Type == uaNs.Type {
 				assert.Equal(a.t, ns.Revision, uaNs.Revision)
+				assert.Equal(a.t, strings.ToLower(fmt.Sprintf("%s-%s-%s", tier.Name, ns.Type, ns.Revision)), uaNs.TemplateRef)
```
Comment (MatousJobanek) [inline pkg/test/masteruserrecord/assertion.go:175]: why not:
```suggestion
				assert.Equal(a.t, ns.TemplateRef, uaNs.TemplateRef)
```

File: pkg/test/masteruserrecord/assertion.go [hunk 2]
```diff
@@ -181,6 +183,7 @@ TierNamespaces:
 		assert.Nil(a.t, userAccount.Spec.NSTemplateSet.ClusterResources)
 	} else {
 		assert.Equal(a.t, tier.Spec.ClusterResources.Revision, userAccount.Spec.NSTemplateSet.ClusterResources.Revision)
+		assert.Equal(a.t, strings.ToLower(fmt.Sprintf("%s-clusterresources-%s", tier.Name, tier.Spec.ClusterResources.Revision)), tier.Spec.ClusterResources.TemplateRef)
```
Comment (MatousJobanek) [inline pkg/test/masteruserrecord/assertion.go:186]: the same here. In addition, the third parameter should be the actual value set in UserAccount/NSTemplateSet
```suggestion
		assert.Equal(a.t, tier.Spec.ClusterResources.TemplateRef, userAccount.Spec.NSTemplateSet.ClusterResources.TemplateRef)
```

**Comment by xcoulon** [comment]
> no idea why, but the build is failing for me with:
> 
> ```
> ../../../../pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/discovery/discovery_client.go:29:2: module github.com/googleapis/gnostic@latest (v0.4.1) found, but does not contain package github.com/googleapis/gnostic/OpenAPIv2
> ```

I got this error a few weeks ago as well and I had to include an extra `replace` directive in `go.mod` in host-operator and member-operator repositories. I'm not sure why I don't get this error here, maybe because I'm using Go 1.14?
Anyways, I'll add the `replace` here too.

**Comment by xcoulon** [comment]
> > no idea why, but the build is failing for me with:
> > ```
> > ../../../../pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/discovery/discovery_client.go:29:2: module github.com/googleapis/gnostic@latest (v0.4.1) found, but does not contain package github.com/googleapis/gnostic/OpenAPIv2
> > ```
> 
> I got this error a few weeks ago as well and I had to include an extra `replace` directive in `go.mod` in host-operator and member-operator repositories. I'm not sure why I don't get this error here, maybe because I'm using Go 1.14?
> Anyways, I'll add the `replace` here too.

done in b46f943

File: pkg/test/masteruserrecord/assertion.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package masteruserrecord
 import (
 	"context"
 	"fmt"
+	"strings"
```
Comment (MatousJobanek) [inline pkg/test/masteruserrecord/assertion.go:6]: unused
```suggestion
```

**Comment by MatousJobanek** [comment]
good catch with the `WithNs` and `WithClusterRes` functions - I missed that.
btw, there is another function using the NSTemplateSet:
https://github.com/codeready-toolchain/toolchain-common/blob/d799dbac939847200c01f7ef81e2d27744bd30d4/pkg/test/masteruserrecord/master_user_record.go#L140-L153
I guess that you should change that too

**Comment by xcoulon** [comment]
> good catch with the `WithNs` and `WithClusterRes` functions - I missed that.
> btw, there is another function using the NSTemplateSet:
> https://github.com/codeready-toolchain/toolchain-common/blob/d799dbac939847200c01f7ef81e2d27744bd30d4/pkg/test/masteruserrecord/master_user_record.go#L140-L153
> 
> 
> I guess that you should change that too

done in 2ad7b4d

**Pull request merged**